src/test/java/com/petros/bibernate/dao/EntityPersisterTest.java

```diff
@@ -0,0 +1,79 @@
+package com.petros.bibernate.dao;
+
+import com.petros.bibernate.exception.BibernateException;
+import com.petros.bibernate.session.model.Product;
+import org.flywaydb.core.Flyway;
+import org.h2.jdbcx.JdbcDataSource;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.math.BigDecimal;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class EntityPersisterTest {
+    private EntityPersister entityPersister;
+
+    @BeforeEach
+    public void setUp() {
+        // Create an H2 in-memory database and run the Flyway migration
+        JdbcDataSource dataSource = new JdbcDataSource();
+        dataSource.setUrl("jdbc:h2:mem:test;DB_CLOSE_DELAY=-1");
+        dataSource.setUser("sa");
+        dataSource.setPassword("");
+        Flyway.configure().dataSource(dataSource).locations("classpath:db/migration/product-test-data").load().migrate();
+        // Initialize the EntityPersister with the H2 data source
+        entityPersister = new EntityPersister(dataSource);
+    }
+
+    @Test
+    public void testFindById() {
+        // Test the findById method with a known entity ID
```

Comment:
You can use @DisplayName annotation to describe what the test does

Reply:
Agree, fixed